### PR TITLE
IGNITE-22152 Support partition filter in SQL

### PR DESCRIPTION
**Pull request by Pochatkin**
https://issues.apache.org/jira/browse/IGNITE-22152

* Added new system column for all tables with partition number
* This column is read only
* __part new reserved column name

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/LogicalRelImplementor.java [hunk 1]
```diff
@@ -463,6 +463,7 @@ public Node<RowT> visit(IgniteTableScan rel) {
             return new ScanNode<>(ctx, Collections.emptyList());
         }
 
+        // TODO: fix required columns.
```
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/LogicalRelImplementor.java:466]: please add like to JIRA

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java [hunk 2]
```diff
@@ -213,50 +214,68 @@ private static IgniteIndex createSchemaIndex(
...
+                Commons.PART_COL_NAME,
+                false,
+                true,
+                true,
+                true,
+                logicalIndex,
+                NativeTypes.INT32,
+                DefaultValueStrategy.DEFAULT_COMPUTED,
+                () -> {
+                    throw new AssertionError("Implicit primary key is generated by a function");
```
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java:266]: misleading message

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/TableRowConverterFactoryImpl.java [hunk 3]
```diff
@@ -56,19 +62,47 @@ public TableRowConverterFactoryImpl(
...
+            assert columnDescriptor.system();
+
+            virtualColumnFactory = (partId) -> new VirtualColumn(columnDescriptor.logicalIndex(), NativeTypes.INT32, false, partId);
+        }
     }
 
     @Override
     public TableRowConverter create(@Nullable BitSet requiredColumns) {
-        if (requiredColumns == null || requiredColumns.cardinality() == schemaDescriptor.length()) {
+        // TODO: fix this. UpdatableTable must pass the bitset with updatable columns.
```
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/TableRowConverterFactoryImpl.java:80]: please assign JIRA to this TODO

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/sqllogic/ItSqlLogicTest.java [hunk 4]
```diff
@@ -139,7 +139,7 @@
 @Tag(value = "sqllogic")
 @ExtendWith({SystemPropertiesExtension.class, WorkDirectoryExtension.class})
 @WithSystemProperty(key = "IMPLICIT_PK_ENABLED", value = "true")
-@SqlLogicTestEnvironment(scriptsRoot = "src/integrationTest/sql")
+@SqlLogicTestEnvironment(scriptsRoot = "src/integrationTest/sql/basic_queries")
```
Comment (korlov42) [inline modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/sqllogic/ItSqlLogicTest.java:142]: unrelated change

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItCreateTableDdlTest.java [hunk 5]
```diff
@@ -196,6 +198,76 @@ public void implicitColocationColumns() {
...
+
+        assertThrowsSqlException(
+                STMT_VALIDATION_ERR,
+                "Failed to validate query. Column '__part' is reserved name.",
+                () -> sql("ALTER TABLE T0 ADD COLUMN \"__part\" INT")
+        );
+    }
+
+    /**
+     * Check implicit partition column configuration (defined by PK)..
```
Comment (PakhomovAlexander) [inline modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItCreateTableDdlTest.java:232]:
```suggestion
     * Check implicit partition column configuration (defined by PK).
```
